cloud_Azure/terraform/module/examples/multiple_accounts_multiple_resource_group/azure_cli.py

```diff
@@ -0,0 +1,46 @@
+import logging
+import os
+import time
+from typing import Any, Optional
+
+from az.cli import az
+
+log = logging.getLogger(__name__)
+
+
+def az_cli(command: str, max_attempts: int = 1) -> Optional[Any]:
+    """
+    Azure CLI commands issued in a quick succession may fail,
+    eg. when trying to configure a resource that is still being created
+    So, allow optional retry with linear backoff
+    """
+
+    for attempt_no in range(max_attempts):
+        data = _az_cli(command)
+        if data is not None:
+            return data
+        if attempt_no < max_attempts - 1:
+            wait_sec = attempt_no + 1
+            log.debug("Retrying Azure CLI command in %d second(s)...", wait_sec)
+            time.sleep(wait_sec)
+    return None  # all attempts failed
+
+
+def _az_cli(command: str) -> Optional[Any]:
```

Comment:
I don't see any clear benefit from changing the API of the "Azure CLI" call. At minimum, the function needs to provide a more bullet-proof indication of failure of success. There may be situations where a successful Azure CLI command returns `None`.

Reply:
the az(...) call used in this function never returns None, it returns empty dict {} when there was no data returned (CLI outputted empty string "")